docs/features/mcp/README.md

```diff
@@ -0,0 +1,324 @@
+
+- **MCP Clients** (like Claude Code) communicate using the MCP protocol over stdio/JSON-RPC
+- **Local Proxy** (`@automattic/mcp-wordpress-remote`) runs on your machine and translates MCP protocol messages to HTTP requests
+- **WordPress MCP Server** receives HTTP requests and processes them through the WordPress Abilities system
+
+This proxy pattern is commonly used in MCP integrations to bridge protocol differences and handle authentication. The `mcp-wordpress-remote` package acts as a protocol translator, converting the stdio-based MCP communication that clients expect into the HTTP REST API calls that WordPress understands.
+
+**Future Evolution**: While this proxy approach provides a robust foundation, future implementations may explore direct MCP protocol support within WordPress or alternative connection methods as the MCP ecosystem evolves.
+
+### Claude Code Integration
+
+To connect Claude Code to your WooCommerce MCP server:
+
+1. Go to **WooCommerce → Settings → Advanced → REST API**
+2. Create a new API key with "Read/Write" permissions
+3. Configure MCP with your API key using Claude Code:
+
+```bash
+claude mcp add woocommerce_mcp \
+  --env WP_API_URL=https://yourstore.com/wp-json/woocommerce/mcp \
+  --env CUSTOM_HEADERS='{"X-MCP-API-Key": "YOUR_CONSUMER_KEY:YOUR_CONSUMER_SECRET"}' \
+  -- npx -y @automattic/mcp-wordpress-remote@latest
+```
+
+### Manual MCP Client Configuration
+
+For other MCP clients, add this configuration to your MCP settings. This configuration tells the MCP client to run the `mcp-wordpress-remote` proxy locally, which will handle the communication with your WordPress server:
+
+```json
+{
+  "mcpServers": {
+    "woocommerce_mcp": {
+      "type": "stdio",
+      "command": "npx",
+      "args": [
+        "-y",
+        "@automattic/mcp-wordpress-remote@latest"
+      ],
+      "env": {
+        "WP_API_URL": "https://yourstore.com/wp-json/woocommerce/mcp",
+        "CUSTOM_HEADERS": "{\"X-MCP-API-Key\": \"YOUR_CONSUMER_KEY:YOUR_CONSUMER_SECRET\"}"
+      }
+    }
+  }
+}
+```
+
+**Important**: Replace `YOUR_CONSUMER_KEY:YOUR_CONSUMER_SECRET` with your actual WooCommerce API credentials.
+
+## Extending MCP Capabilities
+
+### Adding Custom Abilities
+
+Third-party plugins can register additional abilities using the WordPress Abilities API. Abilities can be implemented in various ways - REST endpoint bridging, direct operations, custom logic, or external integrations. Here's a basic example:
+
+```php
+add_action( 'abilities_api_init', function() {
+    wp_register_ability(
+        'your-plugin/custom-operation',
+        array(
+            'label'       => __( 'Custom Store Operation', 'your-plugin' ),
+            'description' => __( 'Performs a custom store operation.', 'your-plugin' ),
+            'callback'    => 'your_custom_ability_handler',
+            'permission'  => 'manage_woocommerce',
```

Comment:
Is 'permission' a new parameter?  I've only seen 'permission_callback' which takes a function. For example:

```
        'permission_callback' =>function () {
            return current_user_can( 'manage_woocommerce' );
        }
```

Docs: https://github.com/WordPress/abilities-api/blob/trunk/docs/3.registering-abilities.md